cvat/apps/engine/views.py

```diff
@@ -851,16 +859,179 @@ def jobs(self, request, pk):
 
         return Response(serializer.data)
 
+    def _is_data_uploading(self) -> bool:
+        return 'data' in self.action
+
     # UploadMixin method
     def get_upload_dir(self):
         if 'annotations' in self.action:
             return self._object.get_tmp_dirname()
-        elif 'data' in self.action:
+        elif self._is_data_uploading():
             return self._object.data.get_upload_dirname()
         elif 'backup' in self.action:
             return backup.get_backup_dirname()
         return ""
 
+    def _get_tus_input_ordering_metafile_name(self) -> str:
+        return "__tus_input_order__"
+
+    def _get_tus_custom_ordering_metafile_name(self) -> str:
+        return "__tus_custom_order__"
+
+    def _get_tus_metafile_names(self) -> List[str]:
+        return [
+            self._get_tus_input_ordering_metafile_name(),
+            self._get_tus_custom_ordering_metafile_name(),
+        ]
+
+    # UploadMixin method
+    def validate_uploaded_file_name(self, filename: str) -> bool:
+        return (
+            super().validate_uploaded_file_name(filename) and
+            osp.basename(filename) not in self._get_tus_metafile_names()
+        )
+
+    def _read_tus_upload_meta_file(self, path: str) -> Sequence[str]:
+        """
+        Reads the input file and returns its contents.
+        """
+
+        with open(path) as f:
+            return [line.strip() for line in f]
+
+    def _prepare_upload_metafile_entry(self, filename: str) -> str:
+        filename = osp.normpath(filename.strip())
+        upload_dir = self.get_upload_dir()
+        if filename.startswith(upload_dir):
+            filename = osp.relpath(filename, upload_dir)
+        return filename.lstrip("/.")
```

Comment:
What is this meant to do? If a file name starts with a dot, then this will remove it, which seems like a bug.

Reply:
Fixed. This is a validation for safety, because input filenames are user-provided. Moved to serializer.